migtools/oadp-non-admin
# Adjust the Status of NonAdminBackup

Moves Status outside of Spec and adjusts this to reflect
Velero Backup status as well additional Status when the
Spec within NonAdminBackup is not defined.


To test:
Scenario 1:
 1. Create NonAdminBackup without BackupSpec, for example:
```
apiVersion: nac.oadp.openshift.io/v1alpha1
kind: NonAdminBackup
metadata:
  name: example
  namespace: nacproject
spec: {}
```
 2.  Check that the Status contains the desired fields:
```
apiVersion: nac.oadp.openshift.io/v1alpha1
kind: NonAdminBackup
metadata:
  creationTimestamp: '2024-03-06T12:53:27Z'
  generation: 1
  managedFields:
    - apiVersion: nac.oadp.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:spec': {}
      manager: Mozilla
      operation: Update
      time: '2024-03-06T12:53:27Z'
    - apiVersion: nac.oadp.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          .: {}
          'f:failureReason': {}
          'f:phase': {}
      manager: main
      operation: Update
      subresource: status
      time: '2024-03-06T12:53:44Z'
  name: example
  namespace: nacproject
  resourceVersion: '37684386'
  uid: bd21d62c-3e86-49de-a2e4-23d42db94ddd
spec: {}
status:
  failureReason: NonAdminBackup CR does not contain valid VeleroBackupSpec
  phase: FailedValidation

```
Scenario 2:
1. Make sure to include the backupSpec:
```
apiVersion: nac.oadp.openshift.io/v1alpha1
kind: NonAdminBackup
metadata:
  name: example
  namespace: nacproject
spec:
  backupSpec: {}
``` 
2. Save the object and check if the Status was populated correctly based on the Velero Backup Status. It can be Failed, but should reflect what original Velero Backup object contains:
```
apiVersion: nac.oadp.openshift.io/v1alpha1
kind: NonAdminBackup
metadata:
  creationTimestamp: '2024-03-06T12:55:55Z'
  generation: 2
  managedFields:
    - apiVersion: nac.oadp.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:spec':
          .: {}
          'f:backupSpec': {}
      manager: Mozilla
      operation: Update
      time: '2024-03-06T12:55:55Z'
    - apiVersion: nac.oadp.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          .: {}
          'f:expiration': {}
          'f:failureReason': {}
          'f:formatVersion': {}
          'f:phase': {}
          'f:startTimestamp': {}
          'f:version': {}
      manager: main
      operation: Update
      subresource: status
      time: '2024-03-06T12:55:55Z'
    - apiVersion: nac.oadp.openshift.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:spec':
          'f:backupSpec':
            'f:itemOperationTimeout': {}
            'f:metadata': {}
            'f:volumeSnapshotLocations': {}
            'f:snapshotMoveData': {}
            'f:defaultVolumesToFsBackup': {}
            'f:ttl': {}
            'f:csiSnapshotTimeout': {}
            'f:storageLocation': {}
            'f:hooks': {}
      manager: manager
      operation: Update
      time: '2024-03-06T12:55:55Z'
  name: example
  namespace: nacproject
  resourceVersion: '37685280'
  uid: 20b1d693-98e0-4adb-bc76-48a531daa421
spec:
  backupSpec:
    volumeSnapshotLocations:
      - velero-sample-1
    defaultVolumesToFsBackup: false
    csiSnapshotTimeout: 10m0s
    ttl: 720h0m0s
    itemOperationTimeout: 4h0m0s
    metadata: {}
    storageLocation: velero-sample-1
    hooks: {}
    snapshotMoveData: false
status:
  expiration: '2024-04-05T12:55:55Z'
  failureReason: >-
    unable to get credentials: unable to get key for secret: Secret
    "cloud-credentials" not found
  formatVersion: 1.1.0
  phase: Failed
  startTimestamp: '2024-03-06T12:55:55Z'
  version: 1
```
